src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -231,28 +251,67 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
-          val callOutputs = entries.map { entry =>
-            val value = entry.wdlValue match {
-              case Some(v) => v
-              case _ => throw new WdlExpressionException(s"Could not evaluate call '${matchedCall.name}', because '${entry.key.name}' is undefined")
+      }
+
+      def lookupCall(name: String): Try[WdlObject] = {
+        parentWorkflow.calls.find {_.name == identifierString} match {
+          case Some(matchedCall) =>
+            val futureValue = fetchCallOutputEntries(matchedCall).map {callOutputEntries =>
+              val callOutputsAsMap = callOutputEntries.map { entry =>
+                entry.key.name -> entry.wdlValue
+              }.toMap
+              callOutputsAsMap.find {case (k,v) => v.isEmpty} match {
+                case Some(noneValue) => Failure(new WdlExpressionException(s"Could not evaluate call ${matchedCall.name} because some if its inputs are not defined (i.e. $noneValue._1"))
+                case None => Success(WdlObject(callOutputsAsMap.map {case (k,v) => k -> v.get}))
+              }
             }
-            entry.key.name -> value
-          }
-          WdlObject(callOutputs.toMap)
+            Await.result(futureValue, DatabaseTimeout)
+          case None => Failure(new WdlExpressionException(s"Could not find a call with name '$identifierString'"))
         }
-        Await.result(futureValue, DatabaseTimeout)
       }
+
+      def lookupDeclaration(name: String): Try[WdlValue] = {
+        parentWorkflow.declarations.find {_.name == identifierString} match {
+          case Some(declaration) =>
+            val futureValue = fetchFullyQualifiedName(declaration.fullyQualifiedName).map {
+              case t: Traversable[SymbolStoreEntry] if t.isEmpty =>
+                Failure(new WdlExpressionException(s"Could not find a declaration with name '$identifierString'"))
+              case t: Traversable[SymbolStoreEntry] if t.size > 1 =>
+                Failure(new WdlExpressionException(s"Expected only one declaration for '$identifierString', got ${t.size}"))
+              case t: Traversable[SymbolStoreEntry] => t.head.wdlValue match {
+                case Some(value) =>
+                  Success(value)
+                case None =>
+                  Failure(new WdlExpressionException(s"No value defined for $identifierString"))
+              }
+            }
+            Await.result(futureValue, DatabaseTimeout)
+          case None =>
+            Failure(new WdlExpressionException(s"Could not find a declaration with name '$identifierString'"))
+        }
+      }
+
+      def notFound = throw new WdlExpressionException(s"Could not resolve $identifierString as a namespace, call, or declaration")
+
+      /* Try each of the three methods in sequence. */
+      val attemptedResolutions = Seq(lookupNamespace _, lookupCall _, lookupDeclaration _).map {_(identifierString)}.find {_.isSuccess}
```

Comment:
This is cool, but a lazy stream of attempted resolutions would stop invoking functions at the first success.